server/cluster/cluster.go

```diff
@@ -349,11 +352,23 @@ func (c *RaftCluster) runServiceCheckJob() {
 
 	checkFn := func() {
 		if c.isAPIServiceMode {
-			once.Do(c.initSchedulers)
-			c.independentServices.Store(mcsutils.SchedulingServiceName, true)
-			return
-		}
-		if c.startSchedulingJobs() {
+			servers, err := discovery.Discover(c.etcdClient, strconv.FormatUint(c.clusterID, 10), mcsutils.SchedulingServiceName)
+			if err != nil {
+				return
+			}
+			if len(servers) != 0 {
+				c.stopSchedulingJobs()
+				once.Do(func() {
+					c.initCoordinator(c.ctx, c, c.hbstreams)
```

Comment:
We should do `initCoordinator` bacause of `prepareChecker`? If so, I think we should do this before `runServiceCheckJob`

Reply:
initCoordinator is used to init scheduler handler